addons/website_form/static/src/snippets/s_website_form/options.js

```diff
@@ -795,10 +865,43 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
      * @override
      */
     onFocus: function () {
+        this._updateAvailableVisibilityDependencies();
+        const dependencyEl = this._getDependency();
+        if (dependencyEl && (['radio', 'checkbox'].includes(dependencyEl.type) || dependencyEl.nodeName === 'SELECT')) {
+            this._updateAvailableVisibilityOptions();
+            return;
+        }
```

Comment:
In the `onRemove`, I make sure that when a field has its visibility depending on the field that is deleted, it becomes visible again.

In the `cleanForSave`, I make sure that all visibility conditions are still consistent. This is not always the case. For example when we change the type of a field, its name changes and the way of composing the visibility condition also changes.
I could make sure that when a field changes its type, the dependent fields are synchronized. This is easy if you change from a text field to a textarea but it is much more complicated if you change from checkbox to number field for example...

Reply:
Why is it more complicated to do it at that time rather than in `cleanForSave` ? :thinking: `cleanForSave` is to do thing you cannot or should not do during edition: for example, `cleanForSave` of a carousel snippet -> go back to first slide of the carousel before saving.

> In the cleanForSave, I make sure that all visibility conditions are still consistent. This is not always the case.

-> What happens if I want to edit a condition that is not consistent then? For me it should always be consistent at any time. This is also better for debugging reason: you always look at something consistent. I am also afraid of the undo/redo behavior if the DOM is sometimes in "inconsistent" states, especially if we ever have the "collaboration edition" feature the editor team is working on.

Reply:
I tried to solve it like you asked me to.
I had no problem with `this._updateAvailableVisibilityDependencies();`

On the other hand for 
`this._updateAvailableVisibilityOptions();`
If I put it in updateUI, it causes a problem...
As I have to do this.rerender=true, updateUI is called again and we end up with an endless loop...

In fact I think I don't understand very well why I have to do this.rerender=true sometimes... Why I don't have to do it for `this._updateAvailableVisibilityDependencies();` but I have to do it for `this._updateAvailableVisibilityOptions();`